Updated Alternate piDSKY2

HerculesFerrari opened the pull request:
I've created new graphics and modified scripts to support 800x480 screens. What I haven't done (yet) is set it up to play nicely with the extant scripts. It also includes support for a neopixel based alarm indicator. How would you like to proceed?

HerculesFerrari committed:
Updated graphics 

New graphics for use with 4.3" 800x480 display and corresponding modified PiDSKY2.py

HerculesFerrari committed:
piDSKY2 for 800x480 screen and nanopixels

Modified piDSKY2 for use with updated higher resolution graphics for use with new fangled higher resolution 4.3" 800x480 displays. Since I'm lazy I've been running these scripts as piDSKY2... but creating instructions to rename files for use seems a tad hacky. I just  want to make this available to the main project because this seems like a wildly underutilized and otherwise nice implementation.

rburkey2005 reviewed (changes requested):
First, great idea and thanks for coming up with this!

I do have a couple of changes I'd like to see instead of committing as-is, though fortunately all of a trivial nature. 

1. Regarding the *.py files:  The headers of the files should be updated.  For one thing, the "Filename" fields should be updated to show the correct filenames (such as "piDSKY2_Prime.py" in place of "piDSKY2.py").  For another, the "Mod history" sections should be updated to show your changes, though exactly how you word those comments is entirely up to you; my personal suggestion would be to remove all of my own comments (such as "2017-11-17 RSB Began") and to instead put in a single comment of your own (such as "2025-07-30 HF Adapted from the file piDSKY2.py").
2. I don't like the naming scheme for the image files:  Just using differing filename extensions (png vs gif) is confusing because it doesn't capture the significant differences between the two sets of identical images (namely, that they have different geometries).  I think it might be better to use a naming scheme like "neo-7Seg-0.png", "neo-7Seg-15.png", etc.  Or "Prime-7Seg-0.png", etc.  (It doesn't have to be either of those.  Those are just examples.)  Or for that matter, leave the names as you have them, but just put them into a separate piDSKY2_Prime-images folder.

Actually, I suppose that using "800x480" instead of "Prime" (e.g., piDSKY2_800x480.py) might be better as well, though I won't insist upon it.

Admittedly, I'm somewhat to blame for these file-naming issues, since I didn't really choose a good file-naming scheme myself to begin with.  But it is what it is.

rburkey2005 commented:
Actually, now that I think about it some more, instead of having a new piDSKY2_Prime.py that supports only 800&times;480 to instead have a piDSKY2.py that supports both the old screen geometry and the new.  I would envisage it having two new command-line switches (say, "--800x480" and "--neo").

rburkey2005 commented:
> That brings runPiDSKY2.sh to the party. Do you have any thoughts on that? Is there an easy way to put a screen resolution selection in the RSET menu, and will that setting become persistent in the cfg?

At the moment, I have to cop out on an answer to this:  I had literally forgotten runPiDSKY2.sh's existence, forgotten that it had a menu, and certainly forgotten all details about how it did whatever it does.  Now that you've reminded me, and I've glanced it at, I do have a vague recollection ... but sadly, you likely know more about it than I do right now.  There will have to be a lot more relearning on my part before I could give any answers that are technically useful.  In general, though, I see no reason that it couldn't be made to do what you are asking for.

Of course, if there were a way that runPiDSKY2.sh or piDSKY2.py could _detect_ the display size and the neopixel-based alarm indicator in an automated way, none of that would be necessary.

HerculesFerrari committed:
Rename 7Seg-0.png to 7Seg-0_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-15.png to 7Seg-15_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-19.png to 7Seg-19_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-21.png to 7Seg-21_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-25.png to 7Seg-25_800x480.png

HerculesFerrari committed:
Rename 7Seg-27.png to 7Seg-27_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-28.png to 7Seg-28_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-29.png to 7Seg-29_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-3.png to 7Seg-3_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-30.png to 7Seg-30_800x480.png

Filename convention update

HerculesFerrari committed:
Rename 7Seg-31.png to 7Seg-31_800x480.png

Filename convention update

HerculesFerrari committed:
Rename CompActyOn.png to CompActyOn_800x480.png

Filename convention update

HerculesFerrari committed:
Rename Dot-gunmetal.png to Dot-gunmetal_800x480.png

Filename convention update

HerculesFerrari committed:
Rename Dot.png to Dot_800x480.png

Filename convention update

HerculesFerrari committed:
Rename MinusOn.png to MinusOn_800x480.png

Filename convention update

HerculesFerrari committed:
Rename NounOn.png to NounOn_800x480.png

Filename convention update

HerculesFerrari committed:
Rename PlusMinusOff.png to PlusMinusOff_800x480.png

Filename convention update

HerculesFerrari committed:
Rename PlusOn.png to PlusOn_800x480.png

Filename convention update

HerculesFerrari committed:
Rename ProgOn.png to ProgOn_800x480.png

Filename convention update

HerculesFerrari committed:
Rename SeparatorOff-gunmetal.png to SeparatorOff-gunmetal_800x480.png

Filename convention update

HerculesFerrari committed:
Rename SeparatorOn.png to SeparatorOn_800x480.png

Filename convention update

HerculesFerrari committed:
Rename VerbOn.png to VerbOn_800x480.png

Filename convention update

HerculesFerrari committed:
Rename SeparatorOn-gunmetal.png to SeparatorOn-gunmetal_800x480.png

Filename convention update

HerculesFerrari committed:
Rename splash.png to splash_800x480.png

Filename convention update

HerculesFerrari committed:
Update and rename piDSKY2_Prime.py to piDSKY2_800x480.py

Changed filename, although it's still a placeholder until new options for switching resolutions are implemented and playing nicely.

rburkey2005 commented:
@HerculesFerrari I won't merge this until all of the changes are completed, but I can't really tell if that has happened yet.  Let me know if/when you're finished.
